funcframework/framework.go

```diff
@@ -293,12 +294,15 @@ func fmtFunctionError(err interface{}) string {
 	return formatted
 }
 
+func sanitizeMessage(message string) string {
+	replacer := strings.NewReplacer("\n", "", "\r", "")
+	return html.EscapeString(replacer.Replace(message))
+}
+
 func writeHTTPErrorResponse(w http.ResponseWriter, statusCode int, status, msg string) {
-	// Ensure logs end with a newline otherwise they are grouped incorrectly in SD.
-	if !strings.HasSuffix(msg, "\n") {
-		msg += "\n"
-	}
-	fmt.Fprint(os.Stderr, msg)
+	// Sanitize message before logging and writing to response
+	escapedMsg := sanitizeMessage(msg)
```

Comment:
We actually need to keep the old behavior here of *adding* a newline to the end of the message. Cloud Logging uses newlines to detect when a new log starts. So if a user didn't add a newline to the end of their function's log like this:

```
log.Print("this doesn't end with a new line")
```

And you invoke the function twice in a row, their Cloud Logging might show something like

`this doesn't end with a new linethis doesn't end with a new line`

When you really want a separate line for each invocation. (Or something similar to that, I dont' remember the exact scenario).